src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -249,95 +280,68 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
 
         const { r0c0, r0c1, r1c0, r1c1 } = this.refs as { [key: string]: HTMLElement };
 
+        const r0c0Table = r0c0.querySelector('table');
+        const r0c1Table = r0c1.querySelector('table');
+        const r1c0Table = r1c0.querySelector('table');
+        const r1c1Table = r1c1.querySelector('table') as HTMLElement;
 
-        // Adjust [fixed columns/fixed rows combo] to fixed rows height
-        let trs = r0c1.querySelectorAll('tr');
-        Array.from(r0c0.querySelectorAll('tr')).forEach((tr, index) => {
-            const tr2 = trs[index];
+        const currentTableWith = getComputedStyle(r1c1Table).width;
 
-            tr.style.height = `${tr2.clientHeight}px`;
-        });
+        if (!cycle) {
+            this.resizeFragmentTable(r0c0Table, currentTableWith);
+            this.resizeFragmentTable(r0c1Table, currentTableWith);
+            this.resizeFragmentTable(r1c0Table, currentTableWith);
+        }
```

Comment:
If not in a cycle, make sure all fragments are the same size as the "main" one (r1c1)